sql/driver/conn.go

```diff
@@ -40,7 +41,7 @@ func (c *conn) Prepare(query string) (driver.Stmt, error) {
 }
 
 func (c *conn) Begin() (driver.Tx, error) {
-	return &tx{conn: c}, nil
+	return &tx{conn: c, txn: []byte{}}, nil
```

Comment:
Do you really need `txn` to be an empty but non-nil byte slice here? Usually a nil byte slice is sufficient.